x/escrow/keeper/keeper.go

```diff
@@ -61,25 +68,51 @@ func (k *keeper) AccountCreate(ctx sdk.Context, id types.AccountID, owner sdk.Ac
 		ID:          id,
 		Owner:       owner.String(),
 		State:       types.AccountOpen,
-		Balance:     deposit,
+		Balance:     sdk.NewCoin(deposit.Denom, sdk.ZeroInt()),
 		Transferred: sdk.NewCoin(deposit.Denom, sdk.ZeroInt()),
 		SettledAt:   ctx.BlockHeight(),
+		Depositor:   depositor.String(),
+		Funds:       sdk.NewCoin(deposit.Denom, sdk.ZeroInt()),
 	}
 
 	if err := obj.ValidateBasic(); err != nil {
 		return err
 	}
 
-	if err := k.bkeeper.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, sdk.NewCoins(deposit)); err != nil {
-		return err
+	switch owner.Equals(depositor) {
+	case true:
+		// use owner's account to add balance to escrow account
+		if err := k.bkeeper.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, sdk.NewCoins(deposit)); err != nil {
+			return err
+		}
+		obj.Balance = deposit
+	case false:
+		if err := k.fetchFundsFromDepositor(ctx, owner, depositor, deposit); err != nil {
+			return err
+		}
+		obj.Funds = deposit
 	}
 
 	store.Set(key, k.cdc.MustMarshal(obj))
 
 	return nil
 }
 
-func (k *keeper) AccountDeposit(ctx sdk.Context, id types.AccountID, amount sdk.Coin) error {
+func (k *keeper) fetchFundsFromDepositor(ctx sdk.Context, owner, depositor sdk.AccAddress, deposit sdk.Coin) error {
+	moduleAddr := k.acctKeeper.GetModuleAddress(types.ModuleName)
+	if moduleAddr == nil {
+		panic(sdkerrors.Wrapf(sdkerrors.ErrUnknownAddress, "module account %s does not exist", types.ModuleName))
```

Comment:
This function can return an error, but also panics. @troian does this make sense?

Reply:
yeah, i think it should be `return sdkerrors.Wrapf(sdkerrors.ErrUnknownAddress, "module account %s does not exist", types.ModuleName)`

Reply:
The panic is legit. The account of a module not existing is a situation that should never happen. `cosmos-sdk` itself handles these as panics. See here: https://github.com/cosmos/cosmos-sdk/blob/055e7e8d80322e4bcac60063245962f8eb06e8d4/x/bank/keeper/keeper.go#L314-L324